src/lib/DataConnectStore.js

```diff
@@ -59,6 +59,22 @@ export default class DataConnectStore {
     return useCase.connectors.map(c1 => this.find(c2 => c1.slug === c2.slug))
   }
 
+  // Fetch all accounts and updates their matching connectors
+  fetchAllAccounts (index) {
+    if (!index && this.accountsIndex) index = this.accountsIndex
```

Comment:
My point should be to offer some methods like:
  * `fetchByName()`
  * `fetchByDate()`
  * `fetchByMetaDataProperty()`

Then each contributor may use the `cozy.client.data` API to create its own as needed, but I think that for basic ones we should encapsulate this logic.

Also, it should not be a blocker for this PR, so let's merge as it and we will have many further occasions to discuss about this aspect :)